gapis/api/vulkan/api/descriptor.api

```diff
@@ -618,25 +618,41 @@ vkCmdBindDescriptorSetsArgs {
 }
 
 sub void dovkCmdBindDescriptorSets(ref!vkCmdBindDescriptorSetsArgs args) {
+   _ = PipelineLayouts[args.Layout]
   dynamic_offset_index := MutableU32(0)
 
-  for i , _ , set in args.DescriptorSets {
-    lastDrawInfo().DescriptorSets[args.FirstSet + as!u32(i)] = DescriptorSets[set]
+  computeInfo := lastComputeInfo()
+  drawInfo := lastDrawInfo()
+
+  currentDescriptorSets := switch args.PipelineBindPoint {
+    case VK_PIPELINE_BIND_POINT_COMPUTE:
+      computeInfo.DescriptorSets
+    case VK_PIPELINE_BIND_POINT_GRAPHICS:
+      drawInfo.DescriptorSets
+  }
+
+  bufferBindingOffsets := switch args.PipelineBindPoint {
+    case VK_PIPELINE_BIND_POINT_COMPUTE:
+      computeInfo.BufferBindingOffsets
+    case VK_PIPELINE_BIND_POINT_GRAPHICS:
+      drawInfo.BufferBindingOffsets
   }
 
   for i in (0 .. len(args.DescriptorSets)) {
     if args.DescriptorSets[as!u32(i)] in DescriptorSets {
-      set := DescriptorSets[args.DescriptorSets[as!u32(i)]]
+      set := args.DescriptorSets[as!u32(i)]
+      setObj := DescriptorSets[set]
+      currentDescriptorSets[args.FirstSet + as!u32(i)] = DescriptorSets[set]
 
       // Since the pDynamicOffsets point into the bindings in order of
       // binding index, and then array index, we have to loop over all
       // of the BoundBuffers in order of the binding number.
 
-      // Since `for _, k, v in set.BoundBuffers` loops over in an
+      // Since `for _, k, v in setObj.BoundBuffers` loops over in an
       // arbitrary order, instead we loop over all bindings for the layout
       // and see if it is a buffer.
-      for j in (0 .. len(set.Bindings)) {
-        binding := set.Bindings[as!u32(j)]
+      for j in (0 .. len(setObj.Bindings)) {
```

Comment:
`setObj.Bindings` is a map of binding number to binding content, not a list of bindings. So for a descriptor set (actually defined in descriptor set layout), the valid bindings might be number 1, 2, 5, for example, then using the `j` here with list looping index: 0, 1, 2, is not correct.